Add missing headers in libpq in module postgres

vivekbagade opened the pull request:
Validations:

Ran `bazel run -- //tools:bcr_validation --check=postgres@14.18.bcr.1`

bazel-io commented:
Hello @UebelAndre, modules you maintain (postgres) have been updated in this PR.
      Please review the changes. You can view a diff against the previous version in the "Generate module diff" check.

vivekbagade commented:
> Why does `include` need to be public? Why isn't depending on `pq` sufficient?

We're trying to bazelify a library like https://github.com/taocpp/taopq and since `pq` uses implementation_deps, some of the needed header files and not accessible.

vivekbagade commented:
> > > Why does `include` need to be public? Why isn't depending on `pq` sufficient?
> > 
> > 
> > We're trying to bazelify a library like https://github.com/taocpp/taopq and since `pq` uses implementation_deps, some of the needed header files and not accessible.
> 
> This target I think includes more than it should for public headers, looking at `libpq-dev_16.9-0ubuntu0.24.04.1_arm64.deb` it seems like the following are available
> 
> ```
> include
> `-- postgresql
>     |-- internal
>     |   |-- c.h
>     |   |-- fe-auth-sasl.h
>     |   |-- libpq
>     |   |   `-- pqcomm.h
>     |   |-- libpq-int.h
>     |   |-- port.h
>     |   |-- postgres_fe.h
>     |   `-- pqexpbuffer.h
>     |-- libpq
>     |   `-- libpq-fs.h
>     |-- libpq-events.h
>     |-- libpq-fe.h
>     |-- pg_config.h
>     |-- pg_config_ext.h
>     |-- pg_config_manual.h
>     |-- pg_config_os.h
>     `-- postgres_ext.h
> ```
> 
> Could you instead add these headers to the `pq` target directly? It would probably also be worth adding these to all versions.

Added the extra headers through LIBPQ_EXTRA and removed the visibility. PTAL

vivekbagade commented:
Note that fe-auth-sasl.h was missing in 14.18 so I didn't add that in

UebelAndre reviewed (dismissed):
> Will do today. Can I make it as part of a different PR? My team is blocked on this change currently

Sure, thanks for the fix!

bazel-io reviewed (dismissed):
Hello @bazelbuild/bcr-maintainers, all modules in this PR have been approved by their maintainers. This PR will be merged if all presubmit checks pass.

vivekbagade commented:
> > Will do today. Can I make it as part of a different PR? My team is blocked on this change currently
> 
> Sure, thanks for the fix!

Resolving
